ipcache: modify labels map in-place in `resolveLabels`

tklauser opened the pull request:
The only caller of `resolveLabels` in `(*IPCache).resolveIdentity` assigns the returned map of labels back to the map of labels passed as an argument, so there is no need to create a copy.

This improves `BenchmarkInjectLabels` as follows:

```
goos: linux
goarch: amd64
pkg: github.com/cilium/cilium/pkg/ipcache
cpu: 13th Gen Intel(R) Core(TM) i7-1365U
                │   old.txt   │              new.txt               │
                │   sec/op    │   sec/op     vs base               │
InjectLabels-12   7.400µ ± 3%   6.857µ ± 1%  -7.34% (p=0.000 n=20)

                │   old.txt    │               new.txt                │
                │     B/op     │     B/op      vs base                │
InjectLabels-12   5.614Ki ± 1%   4.798Ki ± 2%  -14.53% (p=0.000 n=20)

                │  old.txt   │              new.txt              │
                │ allocs/op  │ allocs/op   vs base               │
InjectLabels-12   55.00 ± 0%   53.00 ± 0%  -3.64% (p=0.000 n=20)
```

bimmlerd commented on pkg/ipcache/metadata.go:
```diff
@@ -769,8 +769,6 @@ func (ipc *IPCache) resolveIdentity(ctx context.Context, prefix cmtypes.PrefixCl
 // However, nodes *are* allowed to be selectable by CIDR and CIDR equivalents
 // if PolicyCIDRMatchesNodes() is true.
 func resolveLabels(prefix cmtypes.PrefixCluster, lbls labels.Labels) labels.Labels {
```
If we no longer copy it might make sense to drop the return type here?

tklauser committed:
ipcache: modify labels map in-place in resolveLabels

The only caller of resolveLabels in (*IPCache).resolveIdentity assigns
the returned map of labels back to the map of labels passed as an
argument, so there is no need to create a copy.

This improves BenchmarkInjectLabels as follows:

goos: linux
goarch: amd64
pkg: github.com/cilium/cilium/pkg/ipcache
cpu: 13th Gen Intel(R) Core(TM) i7-1365U
                │   old.txt   │              new.txt               │
                │   sec/op    │   sec/op     vs base               │
InjectLabels-12   7.400µ ± 3%   6.857µ ± 1%  -7.34% (p=0.000 n=20)

                │   old.txt    │               new.txt                │
                │     B/op     │     B/op      vs base                │
InjectLabels-12   5.614Ki ± 1%   4.798Ki ± 2%  -14.53% (p=0.000 n=20)

                │  old.txt   │              new.txt              │
                │ allocs/op  │ allocs/op   vs base               │
InjectLabels-12   55.00 ± 0%   53.00 ± 0%  -3.64% (p=0.000 n=20)

Signed-off-by: Tobias Klauser <tobias@cilium.io>
